packages/backend/src/nest/libp2p/libp2p.service.ts

```diff
@@ -30,15 +32,23 @@ export const LIBP2P_PSK_METADATA = '/key/swarm/psk/1.0.0/\n/base16/\n'
 export class Libp2pService extends EventEmitter {
   public libp2pInstance: Libp2p | null
   public connectedPeers: Map<string, number> = new Map()
+  public dialedPeers: Set<string> = new Set()
+  // public processInChunksService: ProcessInChunks<string>
   private readonly logger = Logger(Libp2pService.name)
   constructor(
     @Inject(SERVER_IO_PROVIDER) public readonly serverIoProvider: ServerIoProviderTypes,
-    @Inject(SOCKS_PROXY_AGENT) public readonly socksProxyAgent: Agent
+    @Inject(SOCKS_PROXY_AGENT) public readonly socksProxyAgent: Agent,
+    private readonly processInChunksService: ProcessInChunksService<string>
   ) {
     super()
   }
 
   private dialPeer = async (peerAddress: string) => {
+    if (this.dialedPeers.has(peerAddress)) {
+      console.log(`Peer ${peerAddress} already dialed, not dialing`) // TODO: remove log
+      return
```

Comment:
How does this work with min/max connections? Once a peer has been dialed, are they never dialed again? Would that mean we might run out of peers to dial if enough peers disconnect?

Reply:
Every dialed peer is added to libp2p's internal address book. Autodialer checks connections every X minutes. Technically (according to code) if number of connections falls below set minConnections (e.g everyone disconnect from us) autodialer should look up peers in libp2p peerStore and try to open connections again.